Project/Assets/_Scripts/_new_Scripts/Weapons/RaycastGunViewModel.cs

```diff
@@ -2,123 +2,141 @@
+            if (model.BulletsInMagazine >= 1 && model.Reloading == false)
+            {
+                // Not waiting for shot to charge and firerate timer
+                if (model.ShotChargeTimer >= model.ShotChargeTime && Time.time > model.FireTimer)
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
 
-    /// <summary>
-    /// Function called on pressing the ReloadButton
-    /// </summary>
-    public override void Reload()
-    {
-        if (model.BulletsInMagazine.Value < model.MagazineSize.Value && model.Reloading.Value == false)
+        public override void StopAttack()
         {
-            StartCoroutine(ExecuteReload());
+            model.ShotChargeTimer = 0f;
         }
-    }
 
-    /// <summary>
-    /// For now Instantiates ray for specified time and destroys it after
-    /// </summary>
-    private IEnumerator ShotEffect()
-    {
-        activeRay = Instantiate(LineRendererPrefab).GetComponent<LineRenderer>();
-        LineRenderer newRay = activeRay;
-        yield return new WaitForSeconds(model.ShotDuration.Value);
-        Destroy(newRay.gameObject);
-    }
+        /// <summary>
+        /// Function called on pressing the ReloadButton
+        /// </summary>
+        public override void Reload()
+        {
+            if (model.BulletsInMagazine < model.MagazineSize && model.Reloading == false)
+            {
+                StartCoroutine(ExecuteReload());
+            }
+        }
 
-    private IEnumerator ExecuteReload()
-    {
-        model.Reloading.Value = true;
-        yield return new WaitForSeconds(model.ReloadTime.Value);
-        model.Reloading.Value = false;
-        model.BulletsInMagazine.Value = model.MagazineSize.Value;
-    }
+        /// <summary>
+        /// For now Instantiates ray for specified time and destroys it after
+        /// </summary>
+        private IEnumerator ShotEffect()
+        {
+            activeRay = Instantiate(LineRendererPrefab).GetComponent<LineRenderer>();
+            LineRenderer newRay = activeRay;
+            yield return new WaitForSeconds(model.ShotDuration);
+            Destroy(newRay.gameObject);
+        }
 
-    private void ExecuteAttack()
-    {
-        model.BulletsInMagazine.Value--;
+        private IEnumerator ExecuteReload()
+        {
+            model.Reloading = true;
+            yield return new WaitForSeconds(model.ReloadTime);
+            model.Reloading = false;
+            model.BulletsInMagazine = model.MagazineSize;
+        }
 
-        for (int i = 0; i < model.BulletCount.Value; ++i)
+        private void ExecuteAttack()
         {
-            model.PenetrationDamage.Value = model.Damage.Value;
-            model.PenetrationLeft.Value = model.BulletPenetration.Value;
+            model.BulletsInMagazine--;
 
-            StartCoroutine(ShotEffect());
+            for (int i = 0; i < model.BulletCount; ++i)
```

Comment:
Dlaczego tutaj iterujesz po ilości naboi w magazynku? 👺

Reply:
To nie ilość naboi w magazynku, tylko ilość wystrzelonych pocisków naraz. Daj znać jak masz lepszą nazwę dla zmiennej :D